esphome/components/analog_threshold/binary_sensor.py

```diff
@@ -0,0 +1,46 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import binary_sensor, sensor
+from esphome.const import (
+    CONF_ID,
+    CONF_SENSOR_ID,
+    CONF_THRESHOLD,
+)
+
+analog_threshold_ns = cg.esphome_ns.namespace("analog_threshold")
+
+AnalogThresholdBinarySensor = analog_threshold_ns.class_(
+    "AnalogThresholdBinarySensor", binary_sensor.BinarySensor, cg.Component
+)
+
+CONF_UPPER = "upper"
+CONF_LOWER = "lower"
+
+CONFIG_SCHEMA = binary_sensor.BINARY_SENSOR_SCHEMA.extend(
+    {
+        cv.GenerateID(): cv.declare_id(AnalogThresholdBinarySensor),
+        cv.Required(CONF_SENSOR_ID): cv.use_id(sensor.Sensor),
+        cv.Required(CONF_THRESHOLD): cv.Any(
+            cv.float_,
+            cv.Schema(
+                {cv.Required(CONF_UPPER): cv.float_, cv.Required(CONF_LOWER): cv.float_}
+            ),
+        ),
+    }
+).extend(cv.COMPONENT_SCHEMA)
+
+
+async def to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    await cg.register_component(var, config)
+    await binary_sensor.register_binary_sensor(var, config)
+
+    bin = await cg.get_variable(config[CONF_SENSOR_ID])
+    cg.add(var.set_sensor(bin))
```

Comment:
Not sure what `bin` means?? No need to use short/confusing names
```suggestion
    sens = await cg.get_variable(config[CONF_SENSOR_ID])
    cg.add(var.set_sensor(sens))
```